src/useDebounce.ts

```diff
@@ -1,22 +1,27 @@
 import { useState, useCallback, useRef, useEffect } from 'react';
 import useDebouncedCallback from './useDebouncedCallback';
 
+function refEquality<T>(left: T, right: T): boolean {
+  return left === right;
+}
+
 export default function useDebounce<T>(
   value: T,
   delay: number,
-  options?: { maxWait?: number; leading?: boolean }
+  options?: { maxWait?: number; leading?: boolean },
+  equalityFn: (left: T, right: T) => boolean = refEquality
```

Comment:
I'd suggest to keep it as a part of the `options` object. Also `equalityFn` isn't nessesary, so we can declare it:
```
equalityFn?: (left: T, right: T) => boolean = refEquality
```

Reply:
I added it to the `options` object. Since I can't make it a default parameter in the signature, I added a check whether `options.equalityFn` is set and then fallback to `valueEquality`.